metadataserver/identity.go

```diff
@@ -52,7 +52,7 @@ func (ms *MetadataServer) generateTaskPodIdentity() (*titus.TaskPodIdentity, err
 	if ms.pod == nil {
 		return nil, fmt.Errorf("No pod object available to marshal as a task identity")
 	}
-	podData, err := proto.Marshal(ms.pod)
+	podData, err := json.Marshal(ms.pod)
```

Comment:
Do you need to use [protojson](https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson) instead of json?

> This package produces a different output than the standard "encoding/json" package, which does not operate correctly on protocol buffer messages.

Reply:
Gosh. I don't know.
That isn't how this pod object is written down on disk in the first place, it is just normal json?

protojson.marshal() requires a proto message in the first place, which I don't have.

I can skip the marshaling step entirely and give you the *literal* bytes on disk, but again, that is just normal json, not protojson.

Reply:
/me so confused

If `ms.pod` isn't a proto message how would `proto.Marshal(ms.pod)` have worked?

Reply:
ms.pod is just a golang object of the pod struct.

`proto.Marshal(ms.pod)` works and turns it into bytes for transport. It somehow knows how to work with any interface (struct)

`json.Marshal(ms.pod)` works because the struct has `json:` annotations so it knows how to make a json byte string.

Keep in mind the json I'm working with on disk is just plain old json:
https://github.com/Netflix/titus-executor/blob/6c1a85add201991ef72ca7e97fda9dc485e77ad1/vk/backend/backend.go#L275

I'm using plain json load to read it in the first place. It can't have that much secret sauce?

Reply:
Got confirmation from @sargun that it won't be possible to use jsonpb in this case, because we don't have the original proto.